examples/checkpoint_example_1.cc

```diff
@@ -91,14 +91,14 @@ int main(int, char**) {
   my_test_inst.a = 10;
   my_test_inst.print();
 
-  auto serialized = checkpoint::serializeType<MyTest>(my_test_inst);
+  auto ret = checkpoint::serialize<MyTest>(my_test_inst);
```

Comment:
Not something for a cleanup PR, but why does this function require the template argument to be specified, rather than inferred?

Reply:
It doesn't actually require the template parameter. Not reason to have it.. will remove.

Reply:
Removed from all examples